CMakeLists.txt

```diff
@@ -99,7 +99,8 @@ option(ENABLE_TEST "Enables unit testing" ON)
 option(ENABLE_PROFILING "Enables profiling support in the application" OFF)
 option(ENABLE_COVERAGE "Enable building and installation of examples" OFF)
 option(ENABLE_EXAMPLES "Enable building and installation of examples" OFF)
-option(ENABLE_DOC "Enable building and installation of documentation" OFF)
+option(ENABLE_USER_DOC "Enable building and installation of the user documentation" OFF)
+option(ENABLE_API_DOC "Enable building and installation of the api documentation" OFF)
```

Comment:
Actually, I would probably still want to guard the doc/ directory with ENABLE_DOC and then the USER_DOC and API_DOC can set that if they are set. If not, ENABLE_DOC should build both targets.